### PR TITLE
Add deny create classic resources policy

### PR DESCRIPTION
**Pull request by mbrat2005**
## Overview/Summary

Adds a Policy assignment using the built-in deny resource type policy which denies the creation of classic resources at the intermediate root management group. 

## This PR fixes/adds/changes/removes

1. Adds policy assignment definition
2. Adds policy assignment reference to eslzArm.json

## Testing Evidence
Confirmed that the policy assignment is created at the intermediate root management group following the portal test, both in dedicated and shared platform infrastructure subscriptions. 

### Testing URLs

The below URLs can be updated where the placeholders are, look for `{YOUR GITHUB BRANCH NAME HERE - Remove Curly Brackets Also}` & `{YOUR GITHUB BRANCH NAME HERE - Remove Curly Brackets Also}`, to allow you to test your portal deployment experience.

> Please also replace the curly brackets on the placeholders `{}`

#### Azure Public

[Deploy To Azure](https://portal.azure.com/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fmbrat2005%2FEnterprise-Scale%2Fadd-deny-create-classic-resources-policy%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2Fmbrat2005%2FEnterprise-Scale%2Fadd-deny-create-classic-resources-policy%2FeslzArm%2Feslz-portal.json)

#### Azure US Gov (Fairfax)
[Deploy To Azure](https://portal.azure.us/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fmbrat2005%2FEnterprise-Scale%2Fadd-deny-create-classic-resources-policy%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2Fmbrat2005%2FEnterprise-Scale%2Fadd-deny-create-classic-resources-policy%2FeslzArm%2Ffairfaxeslz-portal.json)

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [x] Performed testing and provided evidence.
- [x] Ensured [contribution guidance](https://github.com/Azure/Enterprise-Scale/wiki/ALZ-Contribution-Guide) is followed.
- [x] Updated relevant and associated documentation.
- [x] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

### CODE REVIEW

**Review by jtracey93 — CHANGES_REQUESTED** [review]

File: eslzArm/managementGroupTemplates/policyAssignments/DENY-ClassicResourceTypesPolicyAssignment.json [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                            "Microsoft.ClassicStorage/storageAccounts/metrics",
+                            "Microsoft.ClassicStorage/storageAccounts/queueServices",
+                            "Microsoft.ClassicStorage/storageAccounts/services",
+                            "Microsoft.ClassicStorage/storageAccounts/services/diagnosticSettings",
+                            "Microsoft.ClassicStorage/storageAccounts/services/metricDefinitions",
+                            "Microsoft.ClassicStorage/storageAccounts/services/metrics",
+                            "Microsoft.ClassicStorage/storageAccounts/tableServices",
+                            "Microsoft.ClassicStorage/storageAccounts/vmImages",
+                            "Microsoft.ClassicStorage/vmImages",
+                            "Microsoft.ClassicSubscription/operations"
```
[the remark this answers is not in the document]
Comment (mbrat2005) [reply eslzArm/managementGroupTemplates/policyAssignments/DENY-ClassicResourceTypesPolicyAssignment.json:101]: @jtracey93 The built in policy definition for "Not allowed resource types" uses an "in" condition in the rule, which needs to match whole strings.

[the remark this answers is not in the document]
Comment (jtracey93) [reply eslzArm/managementGroupTemplates/policyAssignments/DENY-ClassicResourceTypesPolicyAssignment.json:101]: thanks for confirming and checking

**Review by jtracey93 — CHANGES_REQUESTED** [review]
Thanks @mbrat2005,

- [ ] Could you also update the wiki doc `/docs/wiki/ALZ-Policies.md`?
- [ ] Also do you think we should expose this on the portal experience? @Springstone thoughts? If so, can you support @mbrat2005 in adding this to his PR?

**Comment by jtracey93** [comment]
Deployment succeeded and manually cleaned up - bug #1199 

image

**Comment by mbrat2005** [comment]
> Thanks @mbrat2005,
> 
> * [ ]  Could you also update the wiki doc `/docs/wiki/ALZ-Policies.md`?
> * [ ]  Also do you think we should expose this on the portal experience? @Springstone thoughts? If so, can you support @mbrat2005 in adding this to his PR?

@jtracey93, I think exposing this in the Portal experience makes sense, especially with the Classic VM retirement moving out to September. I'll add it and test.

**Comment by jtracey93** [comment]
> > Thanks @mbrat2005,
> > 
> > * [ ]   Could you also update the wiki doc `/docs/wiki/ALZ-Policies.md`?
> > * [ ]   Also do you think we should expose this on the portal experience? @Springstone thoughts? If so, can you support @mbrat2005 in adding this to his PR?
> 
> @jtracey93, I think exposing this in the Portal experience makes sense, especially with the Classic VM retirement moving out to September.

Yup agreed, maybe expose the radio button on the "Azure Core Setup" blade here:
image

Checkout this file and section for inspiration on where to add this https://github.com/Azure/Enterprise-Scale/blob/main/eslzArm/eslz-portal.json#L82

Let me or @Springstone know if you need any assistance

Thanks @mbrat2005

**Comment by mbrat2005** [comment]
Added Portal control (both to public and fairfax templates). Tested deployment with default 'yes', 'audit' (deploys, action audit), and 'no' (does not deploy).

image

**Review by jtracey93 — APPROVED** [review]
LGTM

**Comment by jtracey93** [comment]
Test ran fine, destroy phase is issue (bug #1199)

**Pull request merged**